core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2687,39 +2685,30 @@ private PhysicalOperation createLookupJoin(
             boolean spillEnabled = isSpillEnabled(session)
                     && node.isSpillable().orElseThrow(() -> new IllegalArgumentException("spillable not yet set"))
                     && !buildOuter;
-            JoinBridgeManager<?> lookupSourceFactory =
-                    createLookupSourceFactoryManager(node, buildNode, buildSymbols, buildHashSymbol, probeSource, context, spillEnabled, localDynamicFilters);
-
-            OperatorFactory operator = createLookupJoin(
```

Comment:
could we keep this method or inline it in separate thread?

Inlining it here diff unnecessary and harder to review

Reply:
It's impossible to keep the method since arguments need to be cast and that is exactly what you wanted to avoid.
Did the inline in a separate commit.